### PR TITLE
Add html-minifier

### PR DESCRIPTION
**Pull request by samnight**

### CODE REVIEW

**Comment by IlyaSemenov** [comment]
Please make this optional and disabled by default. When enabled, I think it should accept the options object so that a user may enable/disable certain options (in particular, `removeComments: true` is not desired in some applications). `true` may mean everything enabled.

File: lib/index.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ const renderTemplate = require('./utils/renderTemplate')
 const emoji = require('./emoji')
 const inspect = require('./utils/inspect')
 const validateConfig = require('./validateConfig')
+const minifyHTML = require('./utils/minifyHTML')
```
Comment (IlyaSemenov) [inline lib/index.js]: Please make it `minifyHtml` not `minifyHTML` (that's a typical convention for abbreviations).

File: lib/utils/minifyHTML.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const { minify } = require('html-minifier')
+
+module.exports = html => {
+  return minify(html, {
```
Comment (IlyaSemenov) [inline lib/utils/minifyHTML.js]: Why the redundant return? You're using ES6 syntax already.

```js
module.exports = html => minify(html, {
 ...
})
```

File: tap-snapshots/examples-custom-root-component-index.test.js-TAP.test.js [hunk 3]
```diff
@@ -8,25 +8,5 @@
...
-  <body>
-    <div id="_ream" data-server-rendered="true"><div data-v-b04334aa><h1 data-v-b04334aa>custom root component</h1> <h1 data-v-b04334aa>homepage</h1></div></div>
-    <script>window.__REAM__={"initialData":{}}</script><script src="/_ream/chunk-vendors.js" defer></script><script src="/_ream/client.js" defer></script>
-  </body>
-</html>
+<!doctype html><html><head><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=no,minimal-ui"><title data-ream-head="true"></title><link rel="preload" href="/_ream/chunk-vendors.js" as="script"><link rel="preload" href="/_ream/client.js" as="script"><link rel="prefetch" href="/_ream/chunk--root.js"><style data-vue-ssr-id="83b416fe:0">h1[data-v-b04334aa]{color:#00f}</style></head><body><div id="_ream" data-server-rendered="true"><div data-v-b04334aa><h1 data-v-b04334aa>custom root component</h1><h1 data-v-b04334aa>homepage</h1></div></div><script>window.__REAM__={"initialData":{}}</script><script src="/_ream/chunk-vendors.js" defer="defer"></script><script src="/_ream/client.js" defer="defer"></script></body></html>
```
Comment (IlyaSemenov) [inline tap-snapshots/examples-custom-root-component-index.test.js-TAP.test.js]: Please keep the existing tests unchanged. They are now unreadable without purpose.

Please add a **new** test instead which tests the new behaviour (or better two, one with default options and one with custom minify options).

**Review by IlyaSemenov — CHANGES_REQUESTED** [review]

**Comment by egoist** [comment]
Not sure if this is safe to do, it might break [Client Side Hydration](https://ssr.vuejs.org/guide/hydration.html)

**Comment by IlyaSemenov** [comment]
The tests now fail on Linux because you forgot to change the module file name casing.

Otherwise looks good to me.

**Comment by SamGold05** [comment]
Sorry, vscode doesn't recognize renaming when change case of filename.

**Comment by SamGold05** [comment]
Now it will be ok.

**Review by IlyaSemenov — APPROVED** [review]

**Comment by IlyaSemenov** [comment]
I published this with a few changes. From the user point of view, I renamed the option to `minifyHtml` (which goes in better accordance with `minimize`).

In the code, I first uniformed the HTML rendering so that minifyHtml could be applied in one place instead of copy/pasting the logic.

**Pull request closed without merging**